### ISSUE TITLE
Equality with a modified copy of the same object does not work

### ISSUE THREAD

**Issue by schatekar** [felangel/equatable#16]
I am using this package mainly for flutter bloc. 

I have a situation where I need to yield the same state but with a field on the state modified. This was failing to trigger the state transition which I traced it down to how this package determines equality. 

I have written a `deepCopy` method that utilises existing constructors on states to create a new copy of the states. I then modify the fields on the copied object. However, since the props are fixed at the time of object creation, if any field is changed after the creation of the object, the `Equatable` base class still uses the field values from when the object was created and results in treating both objects as same. 

I have temporarily fixed this by changing the `props` from a constructor hydrated list to a method that can be overridden and that seems to work for me. Happy to submit a PR.

**Comment by felangel** [comment]
@schatekar can you please provide a simple example app which illustrates the issue you were running into with the bloc library? It would be really useful for me to get more context. Thanks! 👍

**Comment by schatekar** [comment]
@felangel I spent the whole day today to refactor my code as I was not happy with what I was doing. 

My main issue was that instead of yielding a new instance of the state when the new state of of the same type as the old state, I was just modifying the old state and yielding the modified instance. But since the `Equatable` had cached the props during the creation, my modification, later on, did not reach the `Equatable`. 

I changed my implementation, to yield a new instance of the state with the modified field values and that works now. Which is the right way to use the Bloc pattern I guess? So this issue can be closed.
